neqo-http3/src/connection_client.rs

```diff
@@ -5930,4 +5938,113 @@ mod tests {
+        let _ = client.process(datagram, now()).dgram();
+
+        let token = client
+            .events()
+            .find_map(|e| {
+                if let Http3ClientEvent::ResumptionToken(token) = e {
+                    Some(token)
+                } else {
+                    None
+                }
+            })
+            .unwrap();
+
+        // Make a new connection.
+        let mut client = default_http3_client_param(MAX_TABLE_SIZE);
+        assert_eq!(client.state(), Http3State::Initializing);
+        client
+            .enable_resumption(now(), &token)
+            .expect("Set resumption token.");
+
+        assert_eq!(client.state(), Http3State::ZeroRtt);
+        let zerortt_event = |e| matches!(e, Http3ClientEvent::StateChange(Http3State::ZeroRtt));
+        assert!(client.events().any(zerortt_event));
+
+        // Send a request using 0RTT data. This will trigger sending a HEADER_ACK on
+        // the decoder stream before stream id is written.
+        // The client will read HEADER_ACK (value: 0x80) as a stream type. To decode the stream
+        // type we will need 3 more byte (0x80 starts with 10 which is 4 byte encoded verint).
+        // 3 more bytes wil be the corect stream type (0x3) which will be added and to more
+        // requests that will trigger 2 HEADER_ACK. The client will cloed the stream and
+        // the server will trigger closing with error HttpClosedCriticalStream.
+        let _ = make_request(
+            &mut client,
+            true,
+            &[(String::from("myheaders"), String::from("myvalue"))],
+        );
+
+        let out = client.process(None, now());
+        let out = server.process(out.dgram(), now());
+
+        // Send a response
+        let out = client.process(out.dgram(), now());
+        let out = server.process(out.dgram(), now());
+        let out = client.process(out.dgram(), now());
+        let out = server.process(out.dgram(), now());
+        let _ = make_request(
+            &mut client,
+            true,
+            &[(String::from("myheaders2"), String::from("myvalue"))],
+        );
+        let out = client.process(out.dgram(), now());
+        let out = server.process(out.dgram(), now());
+        let _ = make_request(
+            &mut client,
+            true,
+            &[(String::from("myheaders3"), String::from("myvalue"))],
+        );
+        let out = client.process(out.dgram(), now());
+        let out = server.process(out.dgram(), now());
+        let out = client.process(out.dgram(), now());
+        let out = server.process(out.dgram(), now());
+        let out = client.process(out.dgram(), now());
+        let out = server.process(out.dgram(), now());
+        let out = client.process(out.dgram(), now());
+        let out = server.process(out.dgram(), now());
+        let out = client.process(out.dgram(), now());
+        let _ = server.process(out.dgram(), now());
```

Comment:
There are no real assertions here regarding the requests that were made.  You need that to show that the requests are getting through, they are getting processed successfully, and responses are received.